### PR TITLE
Ext_proc refactoring: Move stream object from Filter class to client 

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
This PR is part of the required refactoring needed to support HTTP client in ext_proc: https://github.com/envoyproxy/envoy/issues/35488

It is also to address a comment of https://github.com/envoyproxy/envoy/pull/35740#discussion_r1765336235

### CODE REVIEW

**Comment by yanjunxiang-google** [comment]
@yanavlasov @adisuissa

**Comment by yanjunxiang-google** [comment]
Kind Ping

File: test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc [hunk 1]
```diff
@@ -69,6 +69,14 @@ DEFINE_PROTO_FUZZER(
     return;
   }
 
+  // Limiting the max supported request body size to 128k.
+  if (input.request().has_proto_body()) {
```
Comment (tyxia) [inline test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:73]: Is this needed? 128k is applicable to response not request.

File: source/extensions/filters/http/ext_proc/client_impl.h [hunk 2]
```diff
@@ -32,10 +32,15 @@ class ExternalProcessorClientImpl : public ExternalProcessorClient {
         const Grpc::GrpcServiceConfigWithHashKey& config_with_hash_key,
         const Http::AsyncClient::StreamOptions& options,
         Http::StreamFilterSidestreamWatermarkCallbacks& sidestream_watermark_callbacks) override;
+  ExternalProcessorStream* stream() override { return stream_; }
```
Comment (tyxia) [inline source/extensions/filters/http/ext_proc/client_impl.h:35]: Not a  blocking comment to this PR.

In a follow-up PR, we probably should rename this file to be something like grpc_client_impl, which is aligned with http_client_impl

**Comment by tyxia** [comment]
Also, pass it on to @yanavlasov as this pr is to address https://github.com/envoyproxy/envoy/pull/35740#discussion_r1765336235

File: test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc (hunk 1 above)
Comment (yanjunxiang-google) [reply test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:73]: Thanks for reviewing.

There are an intermittent CI failure: https://dev.azure.com/cncf/envoy/_build/results?buildId=180586&view=logs&j=e969334a-0e55-5c18-ac96-8b546753391e&t=4122df52-7ea9-5745-8c7d-f6b6f2c73149&l=3788
 when run ext_proc_unit_test_fuzz with test case: 
https://github.com/envoyproxy/envoy/blob/aa195eb01d1211aa9e082f2d742d75395975edc0/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_corpus/clusterfuzz-testcase-minimized-ext_proc_unit_test_fuzz-5375162511851520#L16
This test case has a input with request body ~1MB. Such large request body is causing this CI timeout. We can avoid such timeout by add "size = 'large'" in the BUILD file to give this fuzzer target more resource. Or limiting the input body size to a smaller number.  After talking to @adisuissa , it looks to me in fuzzer there is no need to test such a large body, thus added this limit.

File: source/extensions/filters/http/ext_proc/client_impl.h (hunk 2 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/client_impl.h:35]: That's a good idea.  We can make that change either within https://github.com/envoyproxy/envoy/pull/35740, or as a follow up PR. 
I added item here: https://github.com/envoyproxy/envoy/issues/35488#issuecomment-2257374493

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**